pkg/lint/lintersdb/manager.go

```diff
@@ -134,7 +134,7 @@ func (m Manager) GetAllSupportedLinterConfigs() []*linter.Config {
 		goModDirectivesCfg  *config.GoModDirectivesSettings
 		gomodguardCfg       *config.GoModGuardSettings
 		gosecCfg            *config.GoSecSettings
-		gosimpleCfg         *config.StaticCheckSettings
+		gosimpleCfg         config.StaticCheckGoSimpleSettings
```

Comment:
I think it doesn't make sense to use pointers here (for other settings too).
Not big memory win, but code is complicated a lot